bin/reth/src/node/mod.rs

```diff
@@ -191,13 +198,20 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
             dev,
             pruning,
             ext,
+            #[cfg(feature = "optimism")]
+            rollup,
         }
     }
 
     /// Execute `node` command
     pub async fn execute(mut self, ctx: CliContext) -> eyre::Result<()> {
         info!(target: "reth::cli", "reth {} starting", SHORT_VERSION);
 
+        #[cfg(not(any(feature = "ethereum", feature = "optimism")))]
+        compile_error!(
+            "Either feature \"optimism\" or \"ethereum\" must be enabled for this crate."
+        );
```

Comment:
this check should live on the cli level, before any command is selected

Reply:
Agreed, good callout